sqlglot/parser.py

```diff
@@ -2863,7 +2863,11 @@ def _parse_interval(self) -> t.Optional[exp.Interval]:
         if not self._match(TokenType.INTERVAL):
             return None
 
-        this = self._parse_primary() or self._parse_term()
```

Comment:
why not just this = self._parse_term() ?

Reply:
It was breaking at least the following test case:

```python
>>> import sqlglot
>>> sqlglot.transpile("INTERVAL '1 days' * 5")
["INTERVAL ('1 days' * 5)"]
```

The parentheses result in a parse error e.g. in postgres